include/fiction/algorithms/path_finding/distance.hpp

```diff
@@ -62,6 +64,30 @@ template <typename Lyt, typename Dist = double>
     return static_cast<Dist>(std::hypot(x, y));
 }
 
+/**
+ * The Euclidean distance between two SiDBs on the H-Si surface (SiQAD coordinates are required) given by
+ *
+ *  \$ D = \sqrt{(x_1 - x_2)^2 + (y_1 - y_2)^2}$
+ *
+ * @tparam Lyt Coordinate layout type.
+ * @tparam Dist Floating-point type for the distance.
+ * @param lyt Layout.
+ * @param c1 cell coordinate.
+ * @param c2 cell coordinate.
+ * @return Euclidean distance between c1 and c2.
+ */
+template <typename Lyt, typename Dist = double>
+[[nodiscard]] constexpr Dist distance_SiDB_pair([[maybe_unused]] const Lyt& lyt, const cell<Lyt>& c1,
+                                                const cell<Lyt>& c2)
+{
+    static_assert(std::is_same_v<cell<Lyt>, siqad::coord_t>, "coordinate is not a siqad coordinate");
+    const auto pos_c1 = real_position<Lyt>(c1);
+    const auto pos_c2 = real_position<Lyt>(c2);
+    const auto x      = static_cast<double>(pos_c1.first) - static_cast<double>(pos_c2.first);
```

Comment:
Why not just call `euclidean_distance` after you have performed your conversion?